pkgs/build-support/rust/build-rust-crate/configure-crate.nix

```diff
@@ -180,7 +180,10 @@ in ''
          export CARGO_FEATURE_$feature=1
        done
 
-       target/build/${crateName}/build_script_build > target/build/${crateName}.opt
+       # build script failures are reported to stdout rather than stderr, so we
+       # use tee here instead of > to ensure that `nix log` contains useful
+       # information when a failure occurs
+       target/build/${crateName}/build_script_build | tee target/build/${crateName}.opt
```

Comment:
Is this script snippet run with "set -o pipefail"? If it isn't, then the pipe will suppress failures.

Reply:
Thank you for noticing that.  How embarrassing.  What I wrote would have silently omitted features when the build script fails.